content/blog/lift-and-transform-apps-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,283 @@
+
+```yaml
+image:
+  tag: 10.3
+auth:
+  rootPassword: my-password
+global:
+  imagePullSecrets: # Only needed if you're rate limited to Docker Hub
+  - regcred
+primary:
+  persistence:
+    enabled: true
+    storageClass: hpe-transform
+    annotations:
+      csi.hpe.com/importVolAsClone: virt-mm-db-1.vmdk
+    size: 64Gi
+    subPath: lib
+```
+
+The above parameters are quite self-explanatory, the `subPath` directive instructs the kubelet to mount the `lib` directory in the root of the volume. As a result, `/lib` in the volume will be mounted at `/bitnami/mariadb` and the `data` subdirectory which hold the MariaDB databases will be seen properly by the container binaries.
+
+Deploying the chart is straight forward.
+
+```markdown
+$ helm repo add bitnami https://charts.bitnami.com/bitnami
+$ helm install my-test-clone bitnami/mariadb -f values.yaml
+```
+
+Once the chart has been deployed and the database has come up, the “Hello World” test would of course be verifying the database.
+
+```markdown
+$ kubectl exec -it my-test-clone-mariadb-0 -- bash
+I have no name!@my-test-clone-mariadb-0:/$ mysql -ummattsson -ppassword < /bitnami/mariadb/scripts/test_employees_sha.sql
+INFO
+TESTING INSTALLATION
+computation_time
+00:00:49
+summary	result
+CRC	OK
+count	OK
+```
+
+If things dont’ come up properly, which is very likely it won’t on the first couple of iterations, read logs, error messages and course correct accordingly until the database come up cleanly. At this point, in a production scenario, test clients could be attached to further validate and solidify that moving the application to Kubernetes will actually work.
+
+# The endgame
+
+Up until this step, there’s been zero downtime or disruption to the source application. And the last phase is a one-way street. Ensure clients can connect to the destination database instance before proceeding.
+
+While it’s harmless to leave the test clone chart and `PersistentVolumeClaim` on the cluster, we’ll remove it to avoid confusion.
+
+```markdown
+$ helm uninstall my-test-clone
+$ kubectl delete pvc/data-my-test-clone-mariadb-0
+```
+
+Next, we need to prepare the `values-prod.yaml` file.
+
+```yaml
+image:
+  tag: 10.3
+auth:
+  rootPassword: my-password
+global:
+  imagePullSecrets: # Only needed if you're rate limited to Docker Hub
+  - regcred
+primary:
+  persistence:
+    enabled: true
+    storageClass: hpe-transform
+    annotations:
+      csi.hpe.com/importVolumeName: virt-mm-db-1.vmdk
+      csi.hpe.com/forceImport: "true"
+    size: 64Gi
+    subPath: lib
+```
+
+The only change we need to make is the `annotations` stanza. Changing the directive from `importVolAsClone` to `importVolumeName`. We also need to “force” the import as the volume we’re importing already have residual vSphere application metadata associated with it. That will be overwritten by the HPE CSI Driver when forcing the import.
```

Comment:
In the third sentence, change "have" to "has" (it refers to "the volume", which is singular)